pyk/src/pyk/kast.py

```diff
@@ -81,7 +81,16 @@ class KAtt(KAst, Mapping[str, Any]):
     atts: FrozenDict[str, Any]
 
     def __init__(self, atts: Mapping[str, Any] = {}):
-        object.__setattr__(self, 'atts', FrozenDict(atts))
+
+        def _freeze(m: Any) -> Any:
```

Comment:
I do not think think enforcing this here is a good idea. As an analogue, consider `tuple` and `list` -- `FrozenDict` and `dict` are supposed to have the same relationship.

You could introduce a function `freeze` in `pyk.util` though that does this transformation on its argument.

Reply:
You mean I should move the function to `pyk.util`, right? The actual implementation looks OK?

Reply:
Sorry, my mistake. I thought it was an invariant enforced on `FrozenDict` itself. By the way, can a `dict` be a value in an attribute map?

Reply:
Unfortunately it is, :-( for example to store the sort of variables, as seen in this test https://github.com/runtimeverification/k/blob/96f8ea1818875ef64a7ceaaf320c161e0befe3b6/pyk/src/pyk/integration_tests/test_parse_token.py#L23